optimizely/event/event_processor.py

```diff
@@ -121,12 +123,18 @@ def _validate_intantiation_props(self, prop, prop_name):
       False if property name is batch_size and value is a floating point number.
       True otherwise.
     """
-    if (prop_name == 'batch_size' and not isinstance(prop, int)) or prop is None or prop <= 0 or \
-      not validator.is_finite_number(prop):
+    is_valid = True
+
+    if prop is None or not validator.is_finite_number(prop) or prop <= 0:
```

Comment:
The fix part is that **not validator.is_finite_number(prop)** comes before **prop<=0**